crates/rolldown_common/src/types/symbol_ref.rs

```diff
@@ -14,3 +15,55 @@ impl From<(NormalModuleId, SymbolId)> for SymbolRef {
     Self { owner: value.0, symbol: value.1 }
   }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum SymbolOrMemberExprRef {
+  Symbol(SymbolRef),
+  MemberExpr(MemberExprRef),
+}
+
+impl SymbolOrMemberExprRef {
+  /// get the first part of the expression,
+  /// e.g. `test.a.b` will return `test`
+  /// for identifier, it will return itself
+  pub fn first_part(&self) -> &SymbolRef {
+    match self {
+      SymbolOrMemberExprRef::Symbol(s) => s,
+      SymbolOrMemberExprRef::MemberExpr(expr) => &expr.symbol,
+    }
+  }
+}
+
+/// Crossing module ref between symbols
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct MemberExprRef {
```

Comment:
Discussion only: Do you think if it would be better we just call it `SymbolRef`?

```rust
pub struct SymbolRef {
  owner: ...,
  symbol: ...,
  chains: ..., or static_members:...
}

```

Reply:
There is already a struct called `SymbolRef`, do you mean change the original `SymbolRef` sturct? 
the downsides are: 
1. the memory usage will from **8** bytes to **24**
2. the `SymbolRef` will not `Copy` any more

Reply:
Since we used  `SymbolRef` a lot, so the performance difference should be considerable.

Reply:
>  do you mean change the original SymbolRef sturct?

Yeah.

> the memory usage will from 8 bytes to 24
> the SymbolRef will not Copy any more

I'm only think it in aspect of if it make code cleaner. Feel like "Not copy" will make the dx worse. Ok. Let's keep it how it is now and see how it envolves.